pandas/core/categorical.py

```diff
@@ -881,6 +881,54 @@ def notnull(self):
         """
         return ~self.isnull()
 
+    def dropna(self):
+        """
+        Return the Categorical without null values.
+
+        Both missing values (-1 in .codes) and NA as a category are detected.
+        NA is removed from the categories if present.
+
+        Returns
+        -------
+        valid : Categorical
+        """
+        result = self[self.notnull()]
+        if isnull(result.categories).any():
+            result = result.remove_categories([np.nan])
+        return result
+
+    def value_counts(self, dropna=True):
+        """
+        Returns a Series containing counts of each category.
+
+        Every category will have an entry, even those with a count of 0.
+
+        Parameters
+        ----------
+        dropna : boolean, default True
+            Don't include counts of NaN, even if NaN is a category.
+
+        Returns
+        -------
+        counts : Series
+        """
+        import pandas.hashtable as htable
+        from pandas.core.series import Series
+
+        cat = self.dropna() if dropna else self
+        keys, counts = htable.value_count_int64(com._ensure_int64(cat._codes))
+        result = Series(counts, index=keys)
+
+        ix = np.arange(len(cat.categories))
+        if not dropna and -1 in keys:
+            ix = np.insert(ix, 0, -1)
+        result = result.reindex(ix, fill_value=0)
+        result.index = (np.insert(cat.categories, 0, np.nan)
+            if not dropna and -1 in keys
+            else cat.categories)
+
+        return result
+
```

Comment:
I'd like you to fix ``Categorical.describe`` to now use this. The current impl is a hack which does basically this.